numerical_methods/false_position.cpp

```diff
@@ -4,71 +4,93 @@
  * method](https://en.wikipedia.org/wiki/Regula_falsi), also known as the Secant
  * method
  *
- * Given two points \f$a\f$ and \f$b\f$ such that \f$f(a)<0\f$ and
+ * First, multiple intervals are selected with the interval gap provided. 
+ * Separate recursive function called for every root.
+ * Roots are printed Separatelt.
+ *
+ * For an interval [a,b] \f$a\f$ and \f$b\f$ such that \f$f(a)<0\f$ and
  * \f$f(b)>0\f$, then the \f$(i+1)^\text{th}\f$ approximation is given by: \f[
  * x_{i+1} = \frac{a_i\cdot f(b_i) - b_i\cdot f(a_i)}{f(b_i) - f(a_i)}
  * \f]
  * For the next iteration, the interval is selected
  * as: \f$[a,x]\f$ if \f$x>0\f$ or \f$[x,b]\f$ if \f$x<0\f$. The Process is
  * continued till a close enough approximation is achieved.
  *
- * \see newton_raphson_method.cpp, bisection_method.cpp
  */
 #include <cmath>
 #include <cstdlib>
 #include <iostream>
 #include <limits>
 
-#define EPSILON \
-    1e-6  // std::numeric_limits<double>::epsilon()  ///< system accuracy limit
-#define MAX_ITERATIONS 50000  ///< Maximum number of iterations to check
+int RANGE = 100000;  ///< Range in which we have to find the root. (-Range,Range)
+double INTERVAL_GAP = 0.5; /// interval gap. lesser the gap more the accuracy
 
 /** define \f$f(x)\f$ to find root for
  */
-static double eq(double i) {
-    return (std::pow(i, 3) - (4 * i) - 9);  // origial equation
+static double eq(double x) {
+    return (x*x-x);  // original equation
 }
 
-/** get the sign of any given number */
-template <typename T>
-int sgn(T val) {
-    return (T(0) < val) - (val < T(0));
+//recursive regula falsi function for finding root in an interval [x1,x2]
+static double regula_falsi(double x1,double x2,double y1,double y2){
+    double diff = x1-x2;
+    if(diff<0){
+        diff= (-1)*diff;
+    }
+    if(diff<0.00001){          
+        if(y1<0)
+        y1=-y1;
+        if(y2<0)
+        y2=-y2;
+        if(y1<y2)
+        return x1;
+        else
+        return x2;
+    }
+    double x3=0,y3=0;
+    x3 = x1 - (x1-x2)*(y1)/(y1-y2);
+    y3 = eq(x3);
+    return regula_falsi(x2,x3,y2,y3);
 }
 
-/** main function */
-int main() {
-    double a = -1, b = 1, x, z, m, n, c;
-    int i;
-
-    // loop to find initial intervals a, b
-    for (int i = 0; i < MAX_ITERATIONS; i++) {
-        z = eq(a);
-        x = eq(b);
-        if (sgn(z) == sgn(x)) {  // same signs, increase interval
-            b++;
-            a--;
-        } else {  // if opposite signs, we got our interval
-            break;
-        }
+//Function for printing multiple roots
+void printRoot(double root,int COUNT){
```

Comment:
Function documentation is not up to the standards.
Confused? See #962 as a reference. 🙂

Reply:
Function documentation updated as per suggestions.

Reply:
There was problem during batch commit. So, I did all the changes manually.